plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergVariantParquet.java

```diff
@@ -0,0 +1,272 @@
+import static org.apache.iceberg.CatalogUtil.buildIcebergCatalog;
+import static org.apache.iceberg.Files.localOutput;
+import static org.apache.iceberg.jdbc.JdbcCatalog.PROPERTY_PREFIX;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestIcebergVariantParquet
+        extends AbstractTestQueryFramework
+{
+    private JdbcCatalog jdbcCatalog;
+    private File warehouseLocation;
+
+    private static final String INT_COL_NAME = "id";
+    private static final String VARIANT_COL_NAME = "var";
+    private static final Schema SCHEMA = new Schema(
+            Types.NestedField.required(1, INT_COL_NAME, Types.IntegerType.get()),
+            Types.NestedField.required(2, VARIANT_COL_NAME, Types.VariantType.get()));
+    private static final ColumnIdentity INT_COLUMN_IDENTITY = new ColumnIdentity(1, INT_COL_NAME, PRIMITIVE, ImmutableList.of());
+    private static final ColumnIdentity VARIANT_COLUMN_IDENTITY = new ColumnIdentity(2, VARIANT_COL_NAME, VARIANT, ImmutableList.of());
+    private static final GenericRecord RECORD = GenericRecord.create(SCHEMA);
+
+    private static final ByteBuffer TEST_METADATA_BUFFER = VariantTypeTestUtils.createMetadata(ImmutableList.of("a", "b", "c", "d", "e"), true);
+    private static final ByteBuffer TEST_OBJECT_BUFFER = VariantTypeTestUtils.createObject(
+            TEST_METADATA_BUFFER,
+            ImmutableMap.of(
+                    "a", Variants.ofNull(),
+                    "d", Variants.of("trino")));
+    private static final ByteBuffer SIMILAR_OBJECT_BUFFER = VariantTypeTestUtils.createObject(
+            TEST_METADATA_BUFFER,
+            ImmutableMap.of(
+                    "a", Variants.of(123456789),
+                    "c", Variants.of("string")));
+    private static final ByteBuffer EMPTY_OBJECT_BUFFER = VariantTypeTestUtils.createObject(TEST_METADATA_BUFFER, ImmutableMap.of());
+
+    private static final VariantMetadata TEST_METADATA = Variants.metadata(TEST_METADATA_BUFFER);
+    private static final VariantMetadata EMPTY_METADATA = Variants.metadata(VariantTypeTestUtils.emptyMetadata());
+    private static final VariantObject TEST_OBJECT = (VariantObject) Variants.value(TEST_METADATA, TEST_OBJECT_BUFFER);
+    private static final VariantObject SIMILAR_OBJECT = (VariantObject) Variants.value(TEST_METADATA, SIMILAR_OBJECT_BUFFER);
+    private static final VariantObject EMPTY_OBJECT = (VariantObject) Variants.value(TEST_METADATA, EMPTY_OBJECT_BUFFER);
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        warehouseLocation = Files.createTempDirectory("test_iceberg_variant_test").toFile();
+        closeAfterClass(() -> deleteRecursively(warehouseLocation.toPath(), ALLOW_INSECURE));
+        TestingIcebergJdbcServer server = closeAfterClass(new TestingIcebergJdbcServer());
+        jdbcCatalog = (JdbcCatalog) buildIcebergCatalog("tpch", ImmutableMap.<String, String>builder()
```

Comment:
I don't think we want to use JDBC catalog just for testing variant type. It adds unnecessary starting time. We could use file metastore, create a table on Trino, and add a variant column with UpdateSchema class.